### ISSUE TITLE
Langmap breaks python snippets

### ISSUE THREAD

**Issue by Dettorer** [SirVer/ultisnips#431]
I'm using the [bépo keymap](http://bepo.fr/wiki/Accueil) and I rebinded my vim with the following langmap:
````vim
set langmap=$`,\\"1,«2,»3,(4,)5,@6,+7,-8,/9,*0,=-,%=,bq,éw,pe,or,èt,^y,vu,di,lo,jp,z[,w],aa,us,id,ef,\\,g,ch,tj,sk,rl,n\\;,m',ç\\\\,ê<,àz,yx,xc,.v,kb,'n,qm,g\\,,h.,f/,#~,1!,2@,3#,4$,5%,6^,7&,8*,9(,0),°_,`+,BQ,ÉW,PE,OR,ÈT,!Y,VU,DI,LO,JP,Z{,W},AA,US,ID,EF,\\;G,CH,TJ,SK,RL,N:,M\\",Ç\\|,Ê>,ÀZ,YX,XC,:V,KB,?N,QM,G<,H>,F?
set langnoremap
````
Which works pretty well with plugins since [a recent vim update that included the `set langnoremap` option](https://groups.google.com/forum/#!msg/vim_dev/QnNwLWhJ744/1qNcD7d9OvgJ).

I also have a custom UltiSnips snippet for C and C++ header guards:
```python
# Guards
snippet once "Include header once only guard" !
#ifndef ${1:`!p
if not snip.c:
    name = re.sub(r'[^A-Za-z0-9]+','_', snip.fn).upper()
    snip.rv = ('%s' % name).upper()
else:
    snip.rv = snip.c`}
# define $1

${0}

#endif /* !$1 */
endsnippet
```
(My full vim configuration is also available [here](https://bitbucket.org/Dettorer/configuration/src))

Know if I enter the file (for example) `cpio.h` and insert "`once`" then expand the snippet, I get the following result:

Without using the langmap:

```C
#ifndef CPIO_H
# define CPIO_H



#endif /* !CPIO_H */
```
(Which is the expected behaviour)

And with the langmap (and with langnoremap activated):
```C
#ifndef CPIO_H
m+#%oJb~)/>
# define CPIO_H
m+#%oJb~)/>



#endif /* !CPIO_H
m+#%oJb~)/> */
```

Is it a bug in UltiSnips or is `set langnoremap` still not fully functional?

**Comment by SirVer** [comment]
This is likely due to this hack: 

https://github.com/SirVer/ultisnips/blob/master/pythonx/UltiSnips/_vim.py#L237

Maybe it does not work properly with your lang map. The problem here is that UltiSnips needs to select some text and move the cursor around and `feedkeys` did not have the option to ignore the langmap. If that is now available, it would be much better, but my current vim - which is already 7.4 - does not have this option.

Could you try the following: 

1) in the file linke above, at the very bottom, hardcode to use the `_DummyLangMapTranslator` always. Maybe that fixes your issue already. If so, please tell me if there is a `has(langmap)` or similar that we could use to always use that in case it is supported.

2) Please check if feedkeys can be made to also ignore the langmap. That would be an even better solution.

**Comment by SirVer** [comment]
ping?

**Comment by Dettorer** [comment]
Sorry, I didn't see your answer.

Indeed, using the `_DummyLangMapTranslator` in every case at the end fixed it!

To check whether the `langmap`'s behavior is the old one or the one introduced in the patch I linked, you can simply test if the new vim option `langnoremap` is available and was set: with the updated vim, `langmap` has this new behavior only if `set langnoremap` was used.

**Comment by Dettorer** [comment]
Just wanted to make sure everything is alright concerning this issue.
If you need any other information I'm here to help.

**Comment by SirVer** [comment]
No, everything is fine. I just did not find the time to craft a patch yet. If you feel like doing so, please go for it and send a PR. basically the langmap translator should not be used if this option is available.

**Comment by SirVer** [comment]
Fixed in #444
